### ISSUE TITLE
Unable to export S-Function model to FMU with release 3.0 using rtwsfcnfmi TLC

### ISSUE THREAD

**Issue by Gunasekaran145333** [CATIA-Systems/FMIKit-Simulink#424]
I created an S-function using a basic gain block paired with a constant. Using the .mexw64 file in the generate S function block and attempting to build FMU, an error occurs as the call to rtwsfcnfmi_make_rtw_hook, during the after_make hook generated the following error: Failed to build FMU. 

![Generate S Function Block](https://user-images.githubusercontent.com/132982731/237018150-85c7a4bb-0e0f-4cfb-91c6-d0aab9e75eb3.PNG)
![Error](https://user-images.githubusercontent.com/132982731/237018212-1b7f5882-5a5e-43e3-819e-5cd9a48c629c.PNG)

Please help me to resolve this issue

**Comment by TPfitzer** [comment]
You need to read the Info above the marked error:
e.g. in your case the include file 'Trail_sf.h' is missing

**Comment by Gunasekaran145333** [comment]
@TPfitzer . I checked, the file (Trail_sf.h) is already in the path.

**Comment by TPfitzer** [comment]
> @TPfitzer . I checked, the file (Trail_sf.h) is already in the path.

Note its not enough to have it in your matlabpath. The FMIKit will create a new build-folder "..._fmi_..." and all the files needed to be present there. So you need to add them within you simulink configuration or if you have a more complex setup you could use the packNGo feature of matlab.

**Comment by Gunasekaran145333** [comment]
@TPfitzer . In the Simulink configuration, I have included all of the source and header files. However, the same problem continues to appear; I have attached photos of the configuration parameters that I have provided.

![Config1](https://github.com/CATIA-Systems/FMIKit-Simulink/assets/132982731/b7cb85e7-0204-4ab2-8eaa-11006d201345)
![Config2](https://github.com/CATIA-Systems/FMIKit-Simulink/assets/132982731/21455a11-b0d2-47d9-9088-7432e325b5b8)

**Error**
![Error_](https://github.com/CATIA-Systems/FMIKit-Simulink/assets/132982731/4526aca0-96c2-4772-9687-1f65d00db513)


In its essence, we are trying to build FMU for an S-function (Using the .mexw64 file in the generate S function block) using FMI Kit 3.0. Please provide the steps and settings to be kept in Simulink configuration settings. 
It would be great if you could assist me in resolving this.

**Comment by TPfitzer** [comment]
Maybe now i got it you try to build the fmu with only the mex-file and without having the c-code?
If this is the case then i don't know if this is possible at all

**Comment by Gunasekaran145333** [comment]
No, we have source codes for the S-function. It is given in the MATLAB Path as well as in Simulink configuration as you said, but still, it doesn't work. It throws an error as unable to find Subsystem_sf.h file (But is given in both Matlab Path and in Simulink configuration. 

FYI
![Config1](https://github.com/CATIA-Systems/FMIKit-Simulink/assets/132982731/ceac4b5e-0014-4ce8-9cbe-48b4b6dfb314)

**Comment by Gunasekaran145333** [comment]
@TPfitzer, any idea to resolve the above-mentioned issue?
